File: May Leetcode Challenge/Python/905. Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        i = 0
        j = len(nums)-1
        while i<j:
            if nums[i]%2!=0:
                while j>i:
                    if nums[j]%2==0:
                        nums[i], nums[j] = nums[j], nums[i]
                        break
                    else:
                        j-=1
            i+=1
        return nums

File: 905. Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
     int i=0,j=A.size()-1;
        while(i<j)
        {
            if(A[j]%2==0)
            {
                if(A[i]%2!=0)
                {
                    int num=A[j];
                    A[j]=A[i];
                    A[i]=num;
                    i++;
                    j--;
                }
                else
                {
                   i++; 
                }
            }
            else
            {
                j--;
            }
        }
        return A;
    }
};
